[CI] ThreadPoolTests testScheduledFixedDelayRejection failing

This test was introduced in https://github.com/elastic/elasticsearch/pull/106542

**Build scan:**
https://gradle-enterprise.elastic.co/s/vhbjdbibh2se2/tests/:server:test/org.elasticsearch.threadpool.ThreadPoolTests/testScheduledFixedDelayRejection


**Reproduction line:**
```
./gradlew ':server:test' --tests "org.elasticsearch.threadpool.ThreadPoolTests.testScheduledFixedDelayRejection" -Dtests.seed=CA7F3E4D95D9A00F -Dtests.locale=bg-BG -Dtests.timezone=Asia/Anadyr -Druntime.java=22
```

**Applicable branches:**
main

**Reproduces locally?:**
Didn't try

**Failure history:**
[Failure dashboard for `org.elasticsearch.threadpool.ThreadPoolTests#testScheduledFixedDelayRejection`](https://es-delivery-stats.elastic.dev/app/dashboards#/view/dcec9e60-72ac-11ee-8f39-55975ded9e63?_g=(refreshInterval:(pause:!t,value:60000),time:(from:now-7d%2Fd,to:now))&_a=(controlGroupInput:(chainingSystem:HIERARCHICAL,controlStyle:twoLine,ignoreParentSettings:(ignoreFilters:!f,ignoreQuery:!f,ignoreTimerange:!f,ignoreValidations:!t),panels:('0c0c9cb8-ccd2-45c6-9b13-96bac4abc542':(explicitInput:(dataViewId:fbbdc689-be23-4b3d-8057-aa402e9ed0c5,enhancements:(),fieldName:task.keyword,grow:!t,id:'0c0c9cb8-ccd2-45c6-9b13-96bac4abc542',searchTechnique:wildcard,selectedOptions:!(),singleSelect:!t,title:'Gradle%20Task',width:medium),grow:!t,order:0,type:optionsListControl,width:small),'144933da-5c1b-4257-a969-7f43455a7901':(explicitInput:(dataViewId:fbbdc689-be23-4b3d-8057-aa402e9ed0c5,enhancements:(),fieldName:name.keyword,grow:!t,id:'144933da-5c1b-4257-a969-7f43455a7901',searchTechnique:wildcard,selectedOptions:!('testScheduledFixedDelayRejection'),title:Test,width:medium),grow:!t,order:2,type:optionsListControl,width:medium),'4e6ad9d6-6fdc-4fcc-bf1a-aa6ca79e0850':(explicitInput:(dataViewId:fbbdc689-be23-4b3d-8057-aa402e9ed0c5,enhancements:(),fieldName:className.keyword,grow:!t,id:'4e6ad9d6-6fdc-4fcc-bf1a-aa6ca79e0850',searchTechnique:wildcard,selectedOptions:!('org.elasticsearch.threadpool.ThreadPoolTests'),title:Suite,width:medium),grow:!t,order:1,type:optionsListControl,width:medium)))))


**Failure excerpt:**
```
java.lang.AssertionError: safeAwait: CountDownLatch did not reach zero within the timeout

  at __randomizedtesting.SeedInfo.seed([CA7F3E4D95D9A00F]:0)
  at org.junit.Assert.fail(Assert.java:89)
  at org.junit.Assert.assertTrue(Assert.java:42)
  at org.elasticsearch.test.ESTestCase.safeAwait(ESTestCase.java:2097)
  at org.elasticsearch.threadpool.ThreadPoolTests.lambda$blockExecution$12(ThreadPoolTests.java:555)
  at org.elasticsearch.common.util.concurrent.TimedRunnable.doRun(TimedRunnable.java:33)
  at org.elasticsearch.common.util.concurrent.ThreadContext$ContextPreservingAbstractRunnable.doRun(ThreadContext.java:984)
  at org.elasticsearch.common.util.concurrent.AbstractRunnable.run(AbstractRunnable.java:26)
  at java.util.concurrent.ThreadPoolExecutor.runWorker(ThreadPoolExecutor.java:1144)
  at java.util.concurrent.ThreadPoolExecutor$Worker.run(ThreadPoolExecutor.java:642)
  at java.lang.Thread.run(Thread.java:1570)

```

--- a/server/src/test/java/org/elasticsearch/threadpool/ThreadPoolTests.java
+++ b/server/src/test/java/org/elasticsearch/threadpool/ThreadPoolTests.java
@@ -465,7 +465,6 @@ public class ThreadPoolTests extends ESTestCase {
         }
     }
 
-    @AwaitsFix(bugUrl = "https://github.com/elastic/elasticsearch/issues/106618")
     public void testScheduledFixedDelayRejection() {
         final var name = "fixed-bounded";
         final var threadPool = new TestThreadPool(
@@ -476,17 +475,14 @@ public class ThreadPoolTests extends ESTestCase {
         final var future = new PlainActionFuture<Void>();
         final var latch = new CountDownLatch(1);
         try {
+            blockExecution(threadPool.executor(name), latch);
             threadPool.scheduleWithFixedDelay(
-                ActionRunnable.wrap(future, ignored -> Thread.yield()),
+                ActionRunnable.wrap(future, ignored -> fail("should not execute")),
                 TimeValue.timeValueMillis(between(1, 100)),
                 threadPool.executor(name)
             );
 
-            while (future.isDone() == false) {
-                // might not block all threads the first time round if the scheduled runnable is running, so must keep trying
-                blockExecution(threadPool.executor(name), latch);
-            }
-            expectThrows(EsRejectedExecutionException.class, () -> FutureUtils.get(future));
+            expectThrows(EsRejectedExecutionException.class, () -> FutureUtils.get(future, 10, TimeUnit.SECONDS));
         } finally {
             latch.countDown();
             assertTrue(terminate(threadPool));

